test/custom-field-slot-test.html

```diff
@@ -0,0 +1,56 @@
+<!doctype html>
+
+<head>
+  <meta charset="UTF-8">
+  <title>vaadin-custom-field tests</title>
+  <script src="../../web-component-tester/browser.js"></script>
+  <script src="../../webcomponentsjs/webcomponents-lite.js"></script>
+  <link rel="import" href="../../test-fixture/test-fixture.html">
+  <link rel="import" href="../vaadin-custom-field.html">
+</head>
+
+<body>
+  <test-fixture id="default">
+    <template>
+      <vaadin-custom-field>
+        <input type="text">
+        <input type="number">
+      </vaadin-custom-field>
+    </template>
+  </test-fixture>
+
+  <script>
+    describe('slot updates', function() {
+      var customField, inputElement;
+
+      function dispatchSlotChange() {
+        // NOTE(yuriy): Workaround not to use timeouts
```

Comment:
BTW according to `shadydom` tests, this approach might work:

```js
  function afterFlush(fn) {
    if (window.ShadyDOM) {
      ShadyDOM.flush();
      fn();
    } else {
      requestAnimationFrame(fn);
    }
  }
```
See https://github.com/webcomponents/shadydom/blob/6ced5972f8394dcd4acbd16087051b256cadae22/tests/slotchange.html#L54

We could use `async` / `await` for readability, too (needs a small ESLint config update).
Not a blocker but worth checking to reduce usage of this workaround.

Reply:
In most cases will require to wrap the whole test with this method. Currently the same approach with dummy `slotchange` is also used in `vaadin-text-field` (https://github.com/vaadin/vaadin-text-field/blob/e7f0a065e22ec08e892a7d76dfcbca3f0282b709/test/custom-input.html#L78). Will create an improvement ticket and update it in both components if it is Ok.